### ISSUE TITLE
Consider using pydantic aliases

### ISSUE THREAD

**Issue by jthorton** [choderalab/asapdiscovery#387]
I think we might be able to simplify the process of pairing internal schema names to the formatted postera names by using [aliases in pydantic](https://docs.pydantic.dev/2.0/usage/fields/#field-aliases). See the class [here](https://github.com/choderalab/asapdiscovery/blob/8c304d1e86cdfbebe1fa67ce85ab30f06d70c100/asapdiscovery-simulation/asapdiscovery/simulation/szybki.py#L13-L35) for example, to work out what the the postera name is for the variable `szybki_global_strain` we have to check the function [here](https://github.com/choderalab/asapdiscovery/blob/8c304d1e86cdfbebe1fa67ce85ab30f06d70c100/asapdiscovery-simulation/asapdiscovery/simulation/szybki.py#L28C15-L28C15) then we have to map the enum name [here](https://github.com/choderalab/asapdiscovery/blob/8c304d1e86cdfbebe1fa67ce85ab30f06d70c100/asapdiscovery-simulation/asapdiscovery/simulation/szybki.py#L38). Using an alias would allow this to all be defined in a single place like so
```python
from pydantic import BaseModel, Field

class SzybkiFreeformResult(BaseModel):
    """
    Class for storing the results of a Szybki Freeform conformer analysis run.
    """

    ligand_id: str = Field(..., serialization_alias="ligand_id")
    szybki_global_strain: float = Field(..., serialization_alias="ligand-global-strain-szybki-kcal-mol")

c = SzybkiFreeformResult(ligand_id="id_number1", szybki_global_strain="4")
print(c.model_dump(by_alias=True))
```

> {'ligand_id': 'id_number1', 'ligand-global-strain-szybki-kcal-mol': 4.0}

cc @hmacdope

**Comment by hmacdope** [comment]
Yes please! This is a great idea. We will need to update the `renaming` methods to take a schema but I think this is a good change.
